librato/resource_librato_alert.go

```diff
@@ -404,16 +405,25 @@ func resourceLibratoAlertUpdate(d *schema.ResourceData, meta interface{}) error
 		alert.Conditions = conditions
 	}
 	if d.HasChange("attributes") {
-		attributeData := d.Get("attributes").([]interface{})
-		if attributeData[0] == nil {
-			return fmt.Errorf("No attributes found in attributes block")
-		}
-		attributeDataMap := attributeData[0].(map[string]interface{})
-		attributes := new(librato.AlertAttributes)
-		if v, ok := attributeDataMap["runbook_url"].(string); ok && v != "" {
-			attributes.RunbookURL = librato.String(v)
+		v, ok := d.GetOk("attributes")
+
+		// If no attributes are defined, just set to empty attributes.
+		if !ok {
+			attributes := new(librato.AlertAttributes)
+			attributes.RunbookURL = librato.String("")
+			alert.Attributes = attributes
```

Comment:
What about:

```go
alert.Attributes = &librato.AlertAttributes{
  RunbookURL: librato.String(""),
}
```

I'm not totally sure I understand why we want to set RunbookURL to an empty string here. That would suggest that when attributes are not present we send `{"runbook_url": ""}` as the attributes to Librato instead of omitting it or sending an empty object. I'm a bit surprised that that doesn't in turn cause other problems (rejected request as invalid, or the alert always including an empty runbook URL attribute when not set).

Reply:
@bernerdschaefer I believe it doesn't cause an error because of [this](https://github.com/heroku/terraform-provider-librato/blob/e994824b9b4b6bba531eb199512efe7b60d0d330/vendor/github.com/heroku/go-librato/librato/alerts.go#L55) -- ie, if RunbookURL is blank, it doesn't get sent at all (so we send `{}` in that case).

Your method is still better because it avoids `new`, so I'll switch to that and verify.